scipy/spatial/distance.py

```diff
@@ -1195,6 +1199,63 @@ def canberra(u, v, w=None):
     return d
 
 
+def jensenshannon(p, q, base=None):
+    """
+    Compute the Jensen-Shannon divergence between two
+    1-D probability arrays. This is a pseudo metric
+    since it does not satisfy triangle inequality.
+    However, it's squareroot is a metric.
+
+    The Jensen-Shannon divergence between probability
+    vectors `p` and `q` is defined as,
+
+    .. math::
+
+       \\frac{D(p \\parallel m) + D(q \\parallel m)}{2}
+
+    where :math:`m` is the average of :math:`p` and :math:`q`
+    and :math:`D` is the Kullback-Leibler divergence.
+
+    This routine will normalize `p` and `q` if they don't sum to 1.0.
+
+    Parameters
+    ----------
+    p : (N,) array_like
+        left probability vector
+    q : (N,) array_like
+        right probability vector
+    base : double, optional
+        the base of the logarithm used to compute the output
+        if given, then the routine uses the default base of
+        scipy.stats.entropy.
+
+    Returns
+    -------
+    js : double
+        The Jensen-Shannon distance between `p` and `q`
```

Comment:
I think I'm started to get confused -- you're switching between *distance* and *divergence* being returned even within your own docstring -- from the scipy.spatial side of things I'd advocate for providing the distance.